README.md

```diff
@@ -157,6 +157,17 @@ The advanced properties that can be configured for the connector are:
 | client.use.binary.protocol | Set to `false` to force the library to use JSON encoding for REST and realtime operations, instead of msgpack encoding. | *Boolean* | True |
 | client.loglevel | Sets the verbosity of logging. | *Integer* | 0 |
 
+## Dynamic Channel Configuration
+You can configure your channels dynamically by using `#{topic}` and/or `#{key}` placeholders in same configuration values. If you specify `#{key}` in your value, you must send a UTF-8 encoded string as your key.
+Configurations that are supported:
+* `channel`
+  * For example. if you define a channel value with `channel_#{topic}_#{key}` in your configuration, and publish a message to "topic1" with key "key1", the channel will be configured with `channel_topic1_key1`
+* `client.channel.cipher.key`
+  * For example. if you define a `client.channel.cipher.key` value with `client.channel.cipher.key_#{topic}_#{key}` in your configuration, and publish a message to "topic1" with key "key1", the channel will be configured with `client.channel.cipher.key_topic1_key1`
```

Comment:
Add value.

Reply:
Is the value of `client.channel.cipher.key` expected to be the 128/256 bit encryption key? If so, it's unclear why the value would support interpolation.

Reply:
> could we somehow generate a 256 bit key treating the user provided value as passphrase for later work?

I think we have to look at use cases. The encryption key needs to be something the customer can readily access in other systems to decrypt the message in the client, so we need to understand what other systems may be involved, and how keys might be shared between them.

Reply:
> working with generating fixed sized cryptographic keys with given some inputs

The customer could use some form of [key stretching](https://en.wikipedia.org/wiki/Key_stretching) to turn, say, a user password into a cryptographic key, but I think that would be an edge case.

If I were building a system where I wanted per-user encryption keys, I'd generate them randomly, persist them somewhere secure, and make them available over some secure channel to the authenticated user for the duration they need them for. I'd probably expect to code this myself, rather than expecting to reference encryption keys in a config file.

Reply:
>  I'd probably expect to code this myself, rather than expecting to reference encryption keys in a config file.
Agreed.

Reply:
Removed interpolation support for cipher key here. 
https://github.com/ably/kafka-connect-ably/pull/45/commits/d4f2ec84ffc671f3e2ebc344c76cd8f285770c49

There is still one I'm not sure whether I should keep. `client.channel.params ` as this was also a channel specific one but I'm not sure if it makes sense to keep it here. WDYT ?

Reply:
There are two channel parameters: `rewind` and `delta` (see [here](https://ably.com/documentation/realtime/channels/channel-parameters/overview)), neither of which interpolation makes sense for, so I think you can remove interpolation of channel params.

Reply:
https://github.com/ably/kafka-connect-ably/pull/45/commits/0e90a278d3bd4ce960f2b088eb7d0f11d6dcfb33